torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -502,6 +502,40 @@ def _sfdp_replacement_18(query, key, value, causal_mask, dropout_p):
     )
 
 
+def _sfdp_pattern_19(query, key, value, causal_mask, attn_mask, dropout_p):
+    # for token-classification+gpt2 / text-generation+gpt2
+    attn_weights = torch.matmul(query, key.permute(0, 1, 3, 2))
+    inv_scale = torch.full(
+        [],
+        value.size(-1) ** 0.5,
+        dtype=attn_weights.dtype,
```

Comment:
Referring to my comment above: Does this dtype ( in the model we apply this pattern to ) differ from the query tensor dtype in practice?